### ISSUE TITLE
ANSI color programs only look right under Black_on_White

### ISSUE THREAD

**Issue by KLuka** [shellinabox/shellinabox#255]
_From [kveron...@gmail.com](https://code.google.com/u/110579634229550218345/) on January 30, 2014 21:27:23_

What steps will reproduce the problem? 1. Launch a color ncurses program when CSS is set to Black_on_White
2. Notice colors look like proper Term colors.
3. Switch color to White_on_Black
4. Notice how black and white are incorrect.
5. Compare results with standard terminal program What is the expected output? What do you see instead? I expect that regardless if the console is set to White_on_Black, or Black_on_White that ANSI colors are properly disabled regardless. What version of the product are you using? On what operating system? 2.10 on Debian64, package from Downloads tab. Please provide any additional information below. I attempted to tweak the CSS style sheets myself, as ncurses apps run a bit differently.  This appears to be the CSS attrib causing this fuss:
\#vt100 .ansi0

I attempted to add this line, but it didn't work:
\#vt100 #scrollable .ansi0   { color:            #ffffff; }
And updated the other file to:
\#vt100 .ansi0               { color:            `#000000`; }

Unfortunately the browser uses the first style first, I can use the Firefox style inspector to correct it, by unchecking the appropriate box.

The issue is very noticable when a full screen ANSI app has a bright background set, and since ansi0 is also a bright color, the text is unreadable.  Only way to resolve is to change style.  I personally prefer a White_on_Black terminal, and that's the standard on most Linux distributions.

_Original issue: http://code.google.com/p/shellinabox/issues/detail?id=255_

**Comment by KLuka** [comment]
_From [kveron...@gmail.com](https://code.google.com/u/110579634229550218345/) on January 30, 2014 12:44:56_

I found a solution by going through how the DOM and CSS styles are applied, here is a new selectable CSS file, I like to call Green_on_Black(I like it, don't judge):
\#vt100 #scrollable                   { color:            #00ff00;
                                       background-color: `#000000`; }
\#vt100 #scrollable.inverted          { color:            `#000000`;
                                       background-color: #00ff00; }
\#vt100 #scrollable #console .ansi0   { color:            #00ff00; }

When using this, you will need to alter the Color_Terminal.css and change this line:
\#vt100 .ansi0               { color:            `#000000`; }
It may not be required, but I am adding it here just in case.

Anyways, this new CSS file variant will make the ANSI programs render with proper colors when launched, while keeping a nice low-contrast shell.

**Comment by KLuka** [comment]
_From [kveron...@gmail.com](https://code.google.com/u/110579634229550218345/) on January 30, 2014 13:00:05_

It doesn't appear to resolve the issue with "Nano", which also doesn't render correctly under White_on_Black.  There is definitely an issue with rendering ncurses/ANSI enabled programs when not using Black_On_White.

Other example programs are those made with the Newt toolkit, just run their examples while using any style but the default Black_On_White.

**Comment by KLuka** [comment]
_From [evaiba...@gmail.com](https://code.google.com/u/104429237873599617450/) on December 04, 2014 02:36:06_

I ask for your life now bcoz I want product program  call me 09053349601

**Comment by a-detiste** [comment]
I also get an almost not legible white-on-light grey in default `mc` theme, that's why I use the black skin.

![white_on_grey](https://cloud.githubusercontent.com/assets/7994192/7670474/61cdea7a-fca6-11e4-8c40-708e08bbe4ce.png)

**Comment by KLuka** [comment]
I am seeing this too with White on Black color scheme :/ We will have to investigate... Do you have any ideas?

**Comment by KLuka** [comment]
@a-detiste could you check out this PR #311... I think that should fix this issue.

**Comment by a-detiste** [comment]
@KLuka 

That's much better ! :+1: 

There is still this litte problem: in fullscreen program, the background color 'leaks' between each line in Chrome. I never noticed it before because I use `mc`'s dark theme.

![leak1](https://cloud.githubusercontent.com/assets/7994192/7754762/56799c72-ffef-11e4-9e1b-8f0018e8cdc3.png)


I pulled the PR this way, maybe there was a more github-esque way to go:
```
git remote add kluka git@github.com:KLuka/shellinabox.git
git pull kluka
git pull kluka colors
```

**Comment by a-detiste** [comment]
You can already merge this PR and close the bug of course.

**Comment by KLuka** [comment]
@a-detiste Thanks for feedback :+1:

I am also seeing background leaks in all browsers, but only if I use browser zoom (Ctrl + '+') :disappointed: :smile: I will try to find a solution tomorrow, otherwise I will just merge this in and continue some other day...

For pulling PRs I do similar as you. But Github help on the bottom of PR looks useful. I have to try this on next PR :smile: 

```
Step 1: From your project repository, check out a new branch and test the changes.

git checkout -b KLuka-colors master
git pull git://github.com/KLuka/shellinabox.git colors
```

**Comment by KLuka** [comment]
Ok, I merged PR #311 and closed this issue. 

For "leaking background" I still didn't found a solution. In fact I can't even figure out why this is happening :smile: But I think that this is not so critical, and we will solve this someday in the near future :wink:
